internal/framework/provider/example_data_source.go

```diff
@@ -0,0 +1,88 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+)
+
+var _ datasource.DataSource = &ExampleDataSource{}
```

Comment:
I love how the framework leverages interfaces ❤️ 💘

Reply:
Totally. And with Types too, I expect development to be much easier